README.md

```diff
@@ -296,6 +296,28 @@ interface. In case of a missing `interface_name`, the `name` of the profile name
 **Note:** The `name` (the profile name) and the `interface_name` (the device name) may be
 different or the profile may not be tied to an interface at all.
 
+### `interface_path`
+`interface_path` is a list of globbing pathnames(shell wildcard pattern) to match against the
+ID_PATH udev property of devices. udev property ID_PATH represents the persistent path of
+a device. It consists of a subsystem string(pci, usb, platform, etc.) and a subsystem-specific
+identifier. The subsystem-specific identifier for PCI device has format "pci-$domain:$bus:$device.$function",
+where each variable is an hexadecimal value. The path of a device can be obtained with `udevadm
+info /sys/class/net/$dev | grep ID_PATH=` or by looking at the "path" property exported by NetworkManager
+(`nmcli -f general.path device show $dev`). The `interface_path` is optional and restricts the profile
+to be usable only on devices with the given `interface_path`. `interface_path` is only allowed for type
+ethernet or infiniband to match a non-virtual device with the profile.
+
+**Wildcard pattern matching table for `interface_path`:**
+| Pattern       | Meaning                           |
+| ------------- | ----------------------------------|
+| `*`           | matches everything                |
+| `?`           | matches any single character      |
+| `[foo]`       | matches any character in foo      |
+| `|`           | the element is optional, the match evaluates to be true if at least one of the optional element matches (logical OR). By default, an element is optional. This means that an element `foo` behaves the same as `|foo`        |
+| `&`           | the element is mandatory, the match evaluates to be true if all the element matches (logical AND)  |
+| `!`           | an element can also be inverted with exclamation mark (!) between the pipe symbol (or the ampersand) and before the pattern. Note that `!foo` is a shortcut for the mandatory match "&!foo"            |
+| `\`           | a backslash can be used at the beginning of the element (after the optional special characters) to escape the start of the pattern. For example, "&\\!a" is an mandatory match for literally "!a"      |
```

Comment:
this table explains both the wildcard patterns (from `man fnmatch` and `man glob`), but also the special modifiers that NetworkManager has (`&`, `|`, `!`). These are not on the same "level".

first, NetworkManager honors special characters `&`, `|`, `!` and `\`, as explained in the manual page. But these only matter at the beginning of the word!! These are modifiers that affect how a match of the word is used.

in a second step, the remaining words (after removing the special charaters) are used with `fnmatch()` -- where globbing characters are supported.

The table needs to mention that these are different "levels" (however you explain that), and that the modifiers can only be at the beginning of the word.

Reply:
I already described the special modifiers and wildcard pattern at different level, thanks.